### ISSUE TITLE
History of dashboard updates

### ISSUE THREAD

**Issue by costimuraru** [grafana/grafana#1504]
Now that we have the User entity in Grafana 2.0, I think it would be nice to have a change history for a given dashboard. This would help auditing and determine who did what and why. In a big team this could make all the difference. I've often stumbled upon cool dashboards and struggled to find out who created them and discuss why they are needed, how they could be improved and so on. There were also changes to my own dashboards that did not necessarily make sense to me and wanted to find out more from the person who did the changes.

A history like this would help very much:

```
john created dashboard - 19/Feb/15 1:39 PM
costi made changes - 20/Feb/15 1:39 PM - Changed Dashboard title to "..."
costi made changes - 20/Feb/15 2:19 PM - Added Panel "..."
costi made changes - 20/Feb/15 2:39 PM - Updated templating variable "$VAR" to "..."
costi commented on this dashboard - 20/Feb/15 2:40 PM - "Added support for the updated <APP> metrics"
```

I agree that this implies quite an effort, but I think it is something that can be achieved gradually. Once the foundation is in place, we can also contribute with PRs to log different kinds of activities.

**Comment by torkelo** [comment]
I have been thinking the same, a history of who created, and updated etc. What changed can be tricky (unless the user manually adds a comment somewhere on save). 

An extension of this would also be a version history, so you could restore and old version of the dashboard.

**Comment by utkarshcmu** [comment]
This is what I was thinking for in #4086

**Comment by bergquist** [comment]
I really like this idea. I think any larger organization would get lots of value from it. 

Perhaps we should start small. Just keeping a log of who changes a dashboard and at what time?

**Comment by utkarshcmu** [comment]
Perfect :+1

**Comment by mtanda** [comment]
+1

**Comment by chbatey** [comment]
Having a user include a message that is saved would be very useful. Having git as a backend for dashboards so this becomes the commit message would be a great feature.

**Comment by lamadon** [comment]
+1

**Comment by costimuraru** [comment]
I really like the wiki-like history added in the latest versions of grafana. Now, it would be really helpful if we could restore one of the previous versions of a dashboard. In this way if we make a mistake and overwrite the dashboard, we can rollback. Also, following the same principle, if a dashboard gets deleted, we should be able to restore it.

![Uploading grafana.jpg…]()

**Comment by utkarshcmu** [comment]
Dashboard history can be version controlled in Git using wizzy - https://utkarshcmu.github.io/wizzy/

**Comment by jayannah** [comment]
+1

**Comment by kirthiss** [comment]
+1
